src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTag.cs

```diff
@@ -266,13 +266,16 @@ private void Adornment_MouseLeftButtonDown(object sender, MouseButtonEventArgs e
             if (e.ClickCount == 2)
             {
                 e.Handled = true;
-                var replacementValue = _hint.ReplacementTextChange!.Value;
-                var subjectBuffer = _span.Snapshot.TextBuffer;
+                var textChange = _hint.ReplacementTextChange!.Value;
+
+                var snapshot = _span.Snapshot;
+                var subjectBuffer = snapshot.TextBuffer;
 
                 // Selected SpanTrackingMode to be EdgeExclusive by default.
                 // Will revise if there are some scenarios we did not think of that produce undesirable behavior.
-                var currentSnapshotSpan = _span.TranslateTo(subjectBuffer.CurrentSnapshot, SpanTrackingMode.EdgeExclusive);
-                subjectBuffer.Replace(currentSnapshotSpan.Span, replacementValue.NewText);
+                subjectBuffer.Replace(
```

Comment:
I'm assuming the fix here is to use the `textChange.Span.ToSnapshotSpan(...)` instead of `_span`?

Reply:
yup.  _span is hte span of the hint itself in teh document (e.g. where the hint is placed).  But the ReplacementTextChange specifies the span to actually overwrite.